File: HoaCommunityEvents/Domain/Entities/Event.cs
using HoaCommunityEvents.Domain.Common;

namespace HoaCommunityEvents.Domain.Entities;

public class Event : BaseEntity
{
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string Category { get; set; } = string.Empty;
    public string LocationWithinCommunity { get; set; } = string.Empty;
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public int? MaxAttendees { get; set; }
    public string? ImageUrl { get; set; }
    public string HostUserId { get; set; } = string.Empty;
    public string Status { get; set; } = "Pending";

    public AppUser? Host { get; set; }
    public ICollection<EventAttendance> Attendances { get; set; } = new List<EventAttendance>();
}

File: HoaCommunityEvents/frontend/src/types/event.ts
export type HoaEvent = {
  id: string;
  title: string;
  description: string;
  category: string;
  locationWithinCommunity: string;
  startDate: string;
  endDate: string;
  maxAttendees?: number | null;
  imageUrl?: string | null;
  imagePositionX: number;
  imagePositionY: number;
  imageZoom: number;
  hostUserId: string;
  hostDisplayName: string;
  status: string;
  attendeeCount: number;
  isCurrentUserAttending: boolean;
};

export type EventFilter = {
  category?: string;
  status?: string;
  includePending?: boolean;
  sortBy?: string;
  page?: number;
  pageSize?: number;
};

export type PagedResult<T> = {
  items: T[];
  totalCount: number;
  page: number;
  pageSize: number;
  totalPages: number;
};

export type CreateEventFormValues = {
  title: string;
  description: string;
  category: string;
  locationWithinCommunity: string;
  startDate: string;
  endDate: string;
  maxAttendees?: number | null;
  imageUrl?: string | null;
  imagePositionX?: number | null;
  imagePositionY?: number | null;
  imageZoom?: number | null;
};

export type EditEventFormValues = CreateEventFormValues;
